### ISSUE TITLE
Reduce the model filesize if encoder depth is less than 5

### ISSUE THREAD

**Issue by winfried-loetzsch** [qubvel/segmentation_models.pytorch#318]
I noticed that if the depth of an encoder is set to a value less than 5, the unused nodes still remain in the graph and will be saved in checkpoints and torchscript files. This happens for example when the PSPNet is used with standard settings (extracting features after stage 3).

I think it could be beneficial to remove these nodes from the graph for two main reasons:

- the file size of checkpoints and torchscript exports will be much less
- when we debug the model and use print(model) or the like, the output will be easier to understand (not including unused layers)

For the MobileNetV2Encoder I guess the change would look like that:
       
```
def __init__(self, out_channels, depth=5, **kwargs):
        #...

        if self._depth == 4:
            del self.features[14:]
        elif self._depth == 3:
            del self.features[7:]
        elif self._depth == 2:
            del self.features[4:]
        elif self._depth == 1:
            del self.features[2:]
```

**Comment by JulienMaille** [comment]
This is something I mentioned in #171 and which I fixed in my fork if you want to give it a look.
I'll be happy so send a PR to @qubvel if he agrees

**Comment by winfried-loetzsch** [comment]
Yes, I believe it applies to all encoders. The gradients should not be computed as these layers are never executed (at least for mobilenet_v2 I tried). However, the weights are still stored in files and printed during debugging, so deleting them should be the way to go.

**Comment by JulienMaille** [comment]
It seems layers have to be created for pretrained weights to load, and then can be safely deleted.

**Comment by JulienMaille** [comment]
@qubvel is there any plan to clean up unused layers?
